### ISSUE TITLE
Issues around `initializeNetworkHelpers` `TypeError: this.getToken is not a function`

### ISSUE THREAD

**Issue by Pinesy** [Brigad/redux-rest-easy#26]
Regarding the initialization of network helpers, it would seem to me that the `initializeNetworkHelpers` function is not performing in the way it should be. It looks like what is happening is when I pass in a custom config, it inevitably resets itself to the `DEFAULT_NETWORK_HELPERS` config, which for me, still is not working (the `this.getToken is not a function` error occurs with me trying to use an overwritten config, or the default config.

Repo: https://github.com/Pinesy/radical 

The custom config i'm trying to use (just trying to overwrite the `getToken` method)
```
const networkHelpers = {
  getToken: () => sessionStorage.getItem("auth_token")
};
initializeNetworkHelpers(networkHelpers);
```
I am calling `initializeNetworkHelpers` before any network requests being made.

This is an image of the error:
![image](https://user-images.githubusercontent.com/3433722/37163188-1945e18e-22c5-11e8-9b0f-e4b4745656cd.png)

When I click through to the error spot:
![image](https://user-images.githubusercontent.com/3433722/37163220-338d377c-22c5-11e8-8818-21df92a4dd22.png)

Additional info: 
This is the resource I am trying to perform the `retrieve` method on.
```
import { createResource } from "@brigad/redux-rest-easy";

const products = createResource("products")({
  retrieve: {
    method: "GET",
    url: "/products",
    afterHook: () => console.log("Products retrieved successfuly")
  }
});

const {
  actions: { retrieve: { perform: retrieveProducts } },
  selectors: {
    resource: { getResource: getProducts },
    retrieve: { request: { isPerforming: isRetrievingProducts } }
  }
} = products;

export { retrieveProducts, getProducts, isRetrievingProducts };
```

The url is `/products` because in my CRA config, I have my local API proxied, `"proxy": "http://localhost:8080/api"`

**Comment by Zephir77167** [comment]
Hey, thanks for reporting the issue and for the amount of details (and also the repo)!

I am a bit surprised by this, because I am also using it in the same way on a few projects, but have never encountered the error. Also, [this test](https://github.com/Brigad/redux-rest-easy/blob/master/__tests__/internals/network/networkHelpers.test.js#L19) _should_ ensure it doesn't happen.

Nonetheless, try fixing the issue as soon as possible :)

**Comment by Zephir77167** [comment]
The runtime error has been fixed, and a new version has been deployed (2.3.3). Sorry about the inconvenience, and thanks again for your help!

We will soon put live demos in place (using Codesandbox) to prevent this kind of bugs in the future.
